# <span style="color: #20beff;">Bellabeat Case Study (Google Data Analytics Professional Certificate Capstone)</span>

Urška Sršen and Sando Mur founded Bellabeat, a high-tech company that manufactures health-focused smart products. Collecting data on activity, sleep, stress, and reproductive health has allowed Bellabeat to empower women with knowledge about their own health and habits. Since it was founded in .2013, Bellabeat has grown rapidly and quickly positioned  itself as a tech-driven wellness company for women.

By 2016, Bellabeat had opened offices around the world and launched multiple products. Bellabeat products became available through a growing number of online retailers in addition to their own e-commerce channel on their website. The company has invested in traditional advertising media, such as radio, out-of-home billboards, print, and television, but focuses on digital marketing extensively. Bellabeat invests year-round in Google Search, maintaining active Facebook and Instagram pages, and consistently engages consumers on Twitter. Additionally, Bellabeat runs video ads on Youtube and display ads on the Google Display Network to support campaigns around key marketing dates.

Sršen knows that an analysis of Bellabeat’s available consumer data would reveal more opportunities for growth. She has
asked the marketing analytics team to focus on a Bellabeat product and analyze smart device usage data in order to gain
insight into how people are already using their smart devices. Then, using this information, she would like high-level
recommendations for how these trends can inform Bellabeat marketing strategy.

## Scenario

* I am a **Junior Data Analyst** working on the marketing analyst team at Bellabeat, a high-tech manufacturer of health-focused products for women. 
* Bellabeat has the potential to become even larger than what it is in the global smart device market.
* I have been asked to  focus on one of Bellabeat’s products and analyze smart device data to gain insight into how consumers are using their smart devices
* Finally I present my analysis to the executive team along with my recommendations for improving Bellabeat's marketing strategy.


# <span style="color: #20beff;">The complete analysis process</span>

We perform the complete analysis process of this case study based of Google's six steps procedure for data analysis, namely:
* **Ask**: Identifying the business task, considering the key stakeholders
* **Prepare**: Preparing the dataset to be used including checking (ROCCC)
* **Process**: The cleaning, sorting and filtering process to prep the data for calculations and analysis
* **Analyze**: Putting the clean data to work, performing calculations and searching for trends or insights
* **Share**: Communicating your high-level insights and recommendations, this is where visualization comes into action
* **Act**: The final conclusion, how the team and business should apply you insights and what next steps can be taken

# <span style="color: #20beff;">1. ASK Phase </span>

## Business Task
Help Bellabeat make better marketting decisions through analysis of fitness data from non bellabeat smart devices. In our case we use the **FitBit Fitness Tracker Data** 

## Key Stakeholders
* **Urška Sršen**: Bellabeat’s cofounder and Chief Creative Officer
* **Sando Mur**: Mathematician and Bellabeat’s cofounder; key member of the Bellabeat executive team


# <span style="color: #20beff;">2. PREPARE Phase </span>

## Dataset Used:
The dataset used for this case study is the FitBit Fitness Tracker Dataset. 

## Accessibility: 
It is an open-source dataset made available to the public domain.

## Data collection description: 
It was generated by respondents to a distributed survey via Amazon Mechanical Turk between 03.12.2016-05.12.2016. Thirty eligible Fitbit users consented to the submission of personal tracker data, including minute-level output for physical activity, heart rate, and sleep monitoring.

In [1]:
import numpy as np 
import pandas as pd 
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/minuteCaloriesNarrow_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/weightLogInfo_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/minuteMETsNarrow_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/hourlyCalories_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/heartrate_seconds_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/minuteSleep_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/hourlyIntensities_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/hourlySteps_merged.csv
/kaggle/input/fitbit/mturkfitbit_export_3.12.16-4.11.16/Fitabase Data 3.12.16-4.11.16/minuteSteps

## Importing the required datasets

In [2]:
daily_activity = pd.read_csv('/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/dailyActivity_merged.csv')
daily_calories = pd.read_csv('/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/dailyCalories_merged.csv')
daily_steps = pd.read_csv('/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/dailySteps_merged.csv')
daily_intensities = pd.read_csv('/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/dailyIntensities_merged.csv')
daily_sleep = pd.read_csv('/kaggle/input/fitbit/mturkfitbit_export_4.12.16-5.12.16/Fitabase Data 4.12.16-5.12.16/sleepDay_merged.csv')

# <span style="color: #20beff;">3. PROCESS and ANALYSE Phase </span>

## Tools Used: 
I am using Python for this case study due to it's effective data analysis and processing libraries such as pandas. It also supports for great visualization through packages and libraries like Matplotlib and seaborn. I have done the same case study using R as well, which is equally as great in terms of tools for this analysis task and probably more effective for visualization.

In [3]:
daily_activity.info()
daily_activity.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

In [4]:
daily_calories.info()
daily_calories.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Id           940 non-null    int64 
 1   ActivityDay  940 non-null    object
 2   Calories     940 non-null    int64 
dtypes: int64(2), object(1)
memory usage: 22.2+ KB


Id ActivityDay  Calories
0  1503960366   4/12/2016      1985
1  1503960366   4/13/2016      1797
2  1503960366   4/14/2016      1776
3  1503960366   4/15/2016      1745
4  1503960366   4/16/2016      1863

In [5]:
daily_steps.info()
daily_steps.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Id           940 non-null    int64 
 1   ActivityDay  940 non-null    object
 2   StepTotal    940 non-null    int64 
dtypes: int64(2), object(1)
memory usage: 22.2+ KB


Id ActivityDay  StepTotal
0  1503960366   4/12/2016      13162
1  1503960366   4/13/2016      10735
2  1503960366   4/14/2016      10460
3  1503960366   4/15/2016       9762
4  1503960366   4/16/2016      12669

In [6]:
daily_intensities.info()
daily_intensities.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 10 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDay               940 non-null    object 
 2   SedentaryMinutes          940 non-null    int64  
 3   LightlyActiveMinutes      940 non-null    int64  
 4   FairlyActiveMinutes       940 non-null    int64  
 5   VeryActiveMinutes         940 non-null    int64  
 6   SedentaryActiveDistance   940 non-null    float64
 7   LightActiveDistance       940 non-null    float64
 8   ModeratelyActiveDistance  940 non-null    float64
 9   VeryActiveDistance        940 non-null    float64
dtypes: float64(4), int64(5), object(1)
memory usage: 73.6+ KB


Id ActivityDay  SedentaryMinutes  LightlyActiveMinutes  \
0  1503960366   4/12/2016               728                   328   
1  1503960366   4/13/2016               776                   217   
2  1503960366   4/14/2016              1218                   181   
3  1503960366   4/15/2016               726                   209   
4  1503960366   4/16/2016               773                   221   

   FairlyActiveMinutes  VeryActiveMinutes  SedentaryActiveDistance  \
0                   13                 25                      0.0   
1                   19                 21                      0.0   
2                   11                 30                      0.0   
3                   34                 29                      0.0   
4                   10                 36                      0.0   

   LightActiveDistance  ModeratelyActiveDistance  VeryActiveDistance  
0                 6.06                      0.55                1.88  
1                 4.71                      0.69                1.57  
2                 3.91                      0.40                2.44  
3                 2.83                      1.26                2.14  
4                 5.04                      0.41                2.71

In [7]:
daily_sleep.info()
daily_sleep.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  413 non-null    int64 
 1   SleepDay            413 non-null    object
 2   TotalSleepRecords   413 non-null    int64 
 3   TotalMinutesAsleep  413 non-null    int64 
 4   TotalTimeInBed      413 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.3+ KB


Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

We can see that the tables have no null values and `dailly_activities` is a merger of `daily_intensities`, `daily_steps` and `daily_calories` 
We might as well merge the daily sleep to daily activity to create a single table that can keep track of all our daily activities.

In [8]:
value_count = [
    daily_activity['Id'].value_counts().count(), 
    daily_steps['Id'].value_counts().count(),
    daily_calories['Id'].value_counts().count(),
    daily_intensities['Id'].value_counts().count(),
    daily_sleep['Id'].value_counts().count(),
]

table_name = ["daily_activity", "daily_steps", "daily_calories", "daily_intensities", "daily_sleep"]

i = 0;

for x in value_count:
    print("Total contributors in table ", table_name[i], ": ", x)
    i += 1

Total contributors in table  daily_activity :  33
Total contributors in table  daily_steps :  33
Total contributors in table  daily_calories :  33
Total contributors in table  daily_intensities :  33
Total contributors in table  daily_sleep :  24


As we can see entries in the `daily_sleep` table are 24 instead of 33 as in the other cases, so we will have empty cases of sleep values for 9 participants. Our results for sleep related analysis would mostly not hold true for the ones without sleep data. We will deal with the missing values either by dropping the rows all together or by averaging, which ever would suit better.

In [9]:
# merging the daily activity table with the sleep table
merged_daily_activity = pd.merge(daily_activity, daily_sleep, on = 'Id', how = 'left')

In [10]:
merged_daily_activity.info()
merged_daily_activity.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12668 entries, 0 to 12667
Data columns (total 19 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        12668 non-null  int64  
 1   ActivityDate              12668 non-null  object 
 2   TotalSteps                12668 non-null  int64  
 3   TotalDistance             12668 non-null  float64
 4   TrackerDistance           12668 non-null  float64
 5   LoggedActivitiesDistance  12668 non-null  float64
 6   VeryActiveDistance        12668 non-null  float64
 7   ModeratelyActiveDistance  12668 non-null  float64
 8   LightActiveDistance       12668 non-null  float64
 9   SedentaryActiveDistance   12668 non-null  float64
 10  VeryActiveMinutes         12668 non-null  int64  
 11  FairlyActiveMinutes       12668 non-null  int64  
 12  LightlyActiveMinutes      12668 non-null  int64  
 13  SedentaryMinutes          12668 non-null  int64  
 14  Calori

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162            8.5              8.5   
1  1503960366    4/12/2016       13162            8.5              8.5   
2  1503960366    4/12/2016       13162            8.5              8.5   
3  1503960366    4/12/2016       13162            8.5              8.5   
4  1503960366    4/12/2016       13162            8.5              8.5   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.88                      0.55   
2                       0.0                1.88                      0.55   
3                       0.0                1.88                      0.55   
4                       0.0                1.88                      0.55   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 6.06                      0.0                 25   
2                 6.06                      0.0                 25   
3                 6.06                      0.0                 25   
4                 6.06                      0.0                 25   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   13                   328               728      1985   
1                   13                   328               728      1985   
2                   13                   328               728      1985   
3                   13                   328               728      1985   
4                   13                   328               728      1985   

                SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  4/12/2016 12:00:00 AM                1.0               327.0   
1  4/13/2016 12:00:00 AM                2.0               384.0   
2  4/15/2016 12:00:00 AM                1.0               412.0   
3  4/16/2016 12:00:00 AM                2.0               340.0   
4  4/17/2016 12:00:00 AM                1.0               700.0   

   TotalTimeInBed  
0           346.0  
1           407.0  
2           442.0  
3           367.0  
4           712.0

In [11]:
merged_daily_activity.isnull().sum()

Id                            0
ActivityDate                  0
TotalSteps                    0
TotalDistance                 0
TrackerDistance               0
LoggedActivitiesDistance      0
VeryActiveDistance            0
ModeratelyActiveDistance      0
LightActiveDistance           0
SedentaryActiveDistance       0
VeryActiveMinutes             0
FairlyActiveMinutes           0
LightlyActiveMinutes          0
SedentaryMinutes              0
Calories                      0
SleepDay                    227
TotalSleepRecords           227
TotalMinutesAsleep          227
TotalTimeInBed              227
dtype: int64

As we can see above we have about 227 rows with NaN values for the `daily_sleep` table. 